app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.ts

```diff
@@ -0,0 +1,897 @@
+/** @file Module containing the API client for the Cloud backend API.
+ *
+ * Each exported function in the {@link Backend} in this module corresponds to an API endpoint. The
+ * functions are asynchronous and return a `Promise` that resolves to the response from the API. */
+import * as config from '../config'
+import * as http from '../http'
+import * as loggerProvider from '../providers/logger'
+import * as utils from '../utils'
+
+// =================
+// === Constants ===
+// =================
+
+/** Default HTTP body for an "open project" request. */
+const DEFAULT_OPEN_PROJECT_BODY: OpenProjectRequestBody = {
+    forceCreate: false,
+}
+
+/** Relative HTTP path to the "set username" endpoint of the Cloud backend API. */
+const CREATE_USER_PATH = 'users'
+/** Relative HTTP path to the "get user" endpoint of the Cloud backend API. */
+const USERS_ME_PATH = 'users/me'
+/** Relative HTTP path to the "list directory" endpoint of the Cloud backend API. */
+const LIST_DIRECTORY_PATH = 'directories'
+/** Relative HTTP path to the "create directory" endpoint of the Cloud backend API. */
+const CREATE_DIRECTORY_PATH = 'directories'
+/** Relative HTTP path to the "list projects" endpoint of the Cloud backend API. */
+const LIST_PROJECTS_PATH = 'projects'
+/** Relative HTTP path to the "create project" endpoint of the Cloud backend API. */
+const CREATE_PROJECT_PATH = 'projects'
+/** Relative HTTP path to the "list files" endpoint of the Cloud backend API. */
+const LIST_FILES_PATH = 'files'
+/** Relative HTTP path to the "upload file" endpoint of the Cloud backend API. */
+const UPLOAD_FILE_PATH = 'files'
+/** Relative HTTP path to the "create secret" endpoint of the Cloud backend API. */
+const CREATE_SECRET_PATH = 'secrets'
+/** Relative HTTP path to the "list secrets" endpoint of the Cloud backend API. */
```

Comment:
I think I've seen these literals in other place in the code. Are they only here or also somewhere in the cloud repo? If the later, maybe it's worth exporting them to one place and use them from that place? E.g. the cloud repo could import them from this repo? I'm OK with not doing it right now, but I'd love to shortly explore the possible options for the future - what solutions do you see to keep the names synchronized between repos?

Reply:
i *think* they are only here - these are names of cloud endpoints, which *should*, afaict, only be relevant for the cloud frontend

Reply:
I'm 99% sure I've seen `'users/me'` somewhere else. Doesn't really matter now though :)

Reply:
a [search](https://github.com/search?q=repo%3Aenso-org%2Fenso%20users%2Fme&type=code) seems to indicate that it isn't anywhere else at the moment

Reply:
Then I'm totally wrong! Sorry for the confusion!!